src/imitation/scripts/analyze.py

```diff
@@ -1,23 +1,74 @@
 from collections import OrderedDict
+import os
 import os.path as osp
-from typing import Optional
+import tempfile
+from typing import List, Optional
 
 import pandas as pd
 from sacred.observers import FileStorageObserver
+import tensorflow as tf
 
 from imitation.scripts.config.analyze import analysis_ex
 import imitation.util.sacred as sacred_util
 from imitation.util.sacred import dict_get_nested as get
 
 
-@analysis_ex.main
+@analysis_ex.command
+def gather_tb_directories(source_dir: str,
+                          run_name: Optional[str],
+                          env_name: Optional[str],
+                          skip_failed_runs: bool,
+                          ) -> dict:
+  """Gather Tensorboard directories from a `parallel_ex` run.
+  The directories are copied to a unique directory in `/tmp/analysis_tb/` under
+  subdirectories matching the Tensorboard events' Ray Tune trial names.
+
+  Undocumented arguments are the same as in `analyze_imitation()`.
+
+  Args:
+    source_dir: A local_dir for Ray. For example, `~/ray_results/`.
+
+  Returns:
+    A dict with two keys. "gather_dir" (str) is a path to a /tmp/
+    directory containing all the TensorBoard runs filtered from `source_dir`.
+    "n_tb_dirs" (int) is the number of TensorBoard directories that were
+    filtered.
+  """
+  sacred_dicts = _get_sacred_dicts(
+    source_dir, run_name, env_name, skip_failed_runs)
+  os.makedirs("/tmp/analysis_tb", exist_ok=True)
+  tmp_dir = tempfile.mkdtemp(dir="/tmp/analysis_tb/")
+
+  tb_dirs_count = 0
+  for basename in ["tb", "sb_tb"]:
+    os.mkdir(osp.join(tmp_dir, basename))  # Build symlinks in the directory.
+    for sd in sacred_dicts:
+      # Expecting a path like "~/ray_results/{run_name}/sacred/1".
+      # Want to search for all Tensorboard dirs inside
+      # "~/ray_results/{run_name}".
+      sacred_dir = sd.sacred_dir.rstrip("/")
+      run_dir = osp.dirname(osp.dirname(sacred_dir))
+      run_name = osp.basename(run_dir)
+      for tb_src_dir in sacred_util.filter_subdirs(
```

Comment:
This for loops searches for any directory that ends in "tb" or "sb_tb". In other words, it is searching for a Tensorboard directory rather than a Sacred directory. I'll add a comment.

Kind of confusing that filter_subdirs can be used for any type of subdirectory but is in `sacred_util`. Maybe I'll rename `sacred_util` to `file_util` if this happens again.

Reply:
Ah got it, this makes sense.